clients/documents/documents.go

```diff
@@ -275,6 +277,24 @@ func (c Client) create(ctx context.Context, d documents.Document) (api.Response,
 	return processHttpResponse(c.client.Create(ctx, d))
 }
 
+func (c Client) patchWithRetry(ctx context.Context, id string, version int, d documents.Document) (resp api.Response, err error) {
```

Comment:
the underlying core client has a built-in retry mechanism AFAIK (at least it still calls a sendWithRetries method) - no need to reinvent it here, if we setup the options correctly. 

```go
func (c *Client) PATCH(ctx context.Context, endpoint string, body io.Reader, options RequestOptions) (*http.Response, error) {
	return c.sendRequestWithRetries(ctx, http.MethodPatch, endpoint, body, options)
}
```

Reply:
addition: if I'm reading it correctly, the higher level client is intialized with a *rest.Client -> so shouldn't the issue we've run into already be covered via retry options configured on that client? Or does Monaco not set any retry options for the rest client it initializes?